tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py

```diff
@@ -54,8 +59,15 @@ def check_forward(self, cost_data, xs_data, ys_data):
             ys = [self.ys[i][b] for i in range(length)]
             score[b] = self._calc_score(b, ys)
 
-        expect = numpy.sum(-(score - numpy.log(z))) / self.batches[0]
-        testing.assert_allclose(log_p.data, expect)
+        loss = -(score - numpy.log(z))
+        if self.reduce == 'mean':
+            expect = numpy.sum(loss) / self.batches[0]
+            testing.assert_allclose(actual.data, expect)
+        elif self.reduce == 'no':
+            expect = loss
+            self.assertEqual(len(expect), len(actual))
+            for a, e in six.moves.zip(actual, expect):
+                testing.assert_allclose(a.data, e)
```

Comment:
I think `testing.assert_allclose(actual.data, expect)` works is simpler. Is that correct?